lib/PerlIO.pm

```diff
@@ -154,24 +154,120 @@ removed normally.
 
 =item :utf8
 
-Pseudo-layer that declares that the stream accepts Perl's I<internal>
-upgraded encoding of characters, which is approximately UTF-8 on ASCII
-machines, but UTF-EBCDIC on EBCDIC machines.  This allows any character
-Perl can represent to be read from or written to the stream.
-
-This layer (which actually sets a flag on the preceding layer, and is
-implicitly set by any C<:encoding> layer) does not translate or validate
-byte sequences.  It instead indicates that the byte stream will have been
-arranged by other layers to be provided in Perl's internal upgraded
-encoding, which Perl code (and correctly written XS code) will interpret
-as decoded Unicode characters.
-
-B<CAUTION>: Do not use this layer to translate from UTF-8 bytes, as
-invalid UTF-8 or binary data will result in malformed Perl strings.  It is
-unlikely to produce invalid UTF-8 when used for output, though it will
-instead produce UTF-EBCDIC on EBCDIC systems.  The C<:encoding(UTF-8)>
-layer (hyphen is significant) is preferred as it will ensure translation
-between valid UTF-8 bytes and valid Unicode characters.
+Declares that the stream accepts UTF-8. This allows any character perl can
```

Comment:
This seems to be the text from before it was reworked in https://github.com/Perl/perl5/pull/17630. Could you update it to the new text (with any updates as then appropriate)?

Reply:
Done, most of the new text is obsoleted by the changes, but the changes to the first paragraph have been included.